os/dutils.sh

```diff
@@ -57,6 +60,64 @@ function FIND_DBGBINFILE()
 	KERNEL_DBG_BIN="tinyara$EXEEXT"
 }
 
+function TRAP_MENU()
+{
+	echo ======================================================
+	echo "  \"Run TRAP tool for\""
+	echo ======================================================
+	echo "  \"1. Default binary\""
+	echo "  \"2. Custom binary\""
+	echo "  \"x. Exit\""
+	echo ======================================================
+
+	read SELECT_BIN
+	case ${SELECT_BIN,,} in
+		1|default)
+			if [ -z "$(ls -A $BINDIR)" ]; then
+				echo "No output file, Build the code and run TRAP again."
+				exit 1
+			fi
+			echo "Enter the name of the crash log file:"
+			read LOG
+			cp $LOG $TRAPDIR
+			TRAP_RUN log $LOG
+			;;
+		2|manual)
+			echo "Enter the following wrt os directory"
+			echo "For example: ../build/output/bin/ OR bin/"
+			echo ""
+			echo "Enter the binary folder name:"
+			read EXT_BIN_DIR
+			if [ -z "$(ls -A $EXT_BIN_DIR)" ]; then
+				echo "$EXT_BIN_DIR Folder does not exit, try again"
+				exit 1
+			fi
+
+			echo "Enter the configuration file name:"
+			read CONFIG_FILE
+			if [ -z "$(ls -A $CONFIG_FILE)" ]; then
+				echo "$CONFIG_FILE: No such file, try again"
+				exit 1
+			fi
+
+			echo "Enter the Make.defs file name:"
+			read MAKE_FILE
+			if [ -z "$(ls -A $MAKE_FILE)" ]; then
+				echo "$MAKE_FILE: No such file, try again"
+				exit 1
+			fi
```

Comment:
How about we just place the binary, config and make.defs in a predetermined folder with predetermined filenames? Then we dont have to enter the entire path each time.

Reply:
I think keeping it custom is better so that we don't have to mandate user to keep it at one particular place.